src/Stratis.Bitcoin/Consensus/ConsensusManager.cs

```diff
@@ -532,7 +532,7 @@ private async Task<ConnectBlocksResult> FullyValidateLockedAsync(ChainedHeader n
                 // In that case we try to load the blocks from store, if store is not present we disconnect all peers.
                 this.HandleMissingBlocksGap(currentTip);
 
-                var result = new ConnectBlocksResult(false);
+                var result = ConnectBlocksResult.Failed();
```

Comment:
I still find this construct weird.
what was the problem with the previous one?

Reply:
@bokobza the previous constructor had one required parameter and 4 defaulted parameters which I think is quiet messy and not intuitive at all. It is so easy to NOT understand what constructor to use and this way it is more expressive as to what the intent is of the `ConnectBlocksResult`. It aids in helping the dev choose the right constructor via a static one and improves readability.

Reply:
I really don't think it's clear. You have method called Success(), a method called Failed() and a method called `FailedAndBanPeers()` where I'm not sure what it does. Does it ban peers?
Something that I think could be clearer is 

`public ConnectBlocksResult(bool success)` + a bunch of properties.

so you can do 
```
var result = new ConnectBlocksResult(false);
result.banDurationSeconds = 222;
result.BanReason = error.Message;
result.ConsensusTipChanged = false;
result.Error = error;
result.PeersToBan = peersToBan;
```